### PR TITLE
Introduce WORK_IN_PROGRESS clearing state

### LINKED ISSUE #608: Introduce Antenna Internal Clearing State "WIP" for Compliance Tool
### Summary of the Feature
When a release has information collected already, but is not approved yet, in the compliance tool SW360 Updater, it is not possible to update the release with this information without automatically setting the SW360 specific clearing state to "approved". This is problematic, since this comes with a clearing document that can, presumably, not be changed any more. 

I propose a "work in progress" in our clearing state enum, so when running the updater a filter can be used to distinct between `INITIAL`, `WIP` and all other clearing states. If the WIP state is the case, then the release does only get updated without adding a clearing document to the release. 

Additional Suggestion: The Clearing State that is Antenna specific could also be remained to reflect something  like "curation state", or we finally make use of the "Change Status" existing in SW360. The state as is, is problematic-

### Acceptance Criteria
- [ ] Releases can be updated without "approving" them in SW360 via the compliance tool updater
- [ ] those releases can be marked with a state in the csv file, so to avoid to update ALL releases in a csv file even if they haven't changed

### Definition of Done
- Acceptance criteria fulfilled
- A test case is created to reproduce the bug
- A PR is created, the CI infrastructure reports green, the bug test case proves that bug is fixed
- The PR is reviewed and approved
- No TODOs left in the code unless explained in the ticket, if something else is still open, this is summarized in a comment in the issue
- Documentation is updated

### PR DESCRIPTION
**Pull request by KorSin**
Resolves #608

A release can now have a WORK_IN_PROGRESS clearing state.
This updates or creates the release on SW360 but does not upload
any attachment.

Signed-off-by: Korbinian Singhammer <external.Korbinian.Singhammer2@bosch.io>

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to? (*Refer to issue here*)
> * How is it solving the issue?
> * Did you add or update any new dependencies that are required for your change?

@neubs-bsi @oheger-bosch 

*Type of change*:  new feature

### Checklist
Must:
- [x] All related issues are referenced in commit messages

Optional: *(delete if not applicable)*
- [x] I have provided tests for the changes (if there are changes that need additional tests)
- [x] I have updated the documentation accordingly to my changes

### CODE REVIEW

**Comment by KorSin** [comment]
@neubs-bsi, @oheger-bosch In terms of automated tests I am satisfied with the changes. However, manually testing the changes with an SW360 instance leaves me a bit confused. If I update a not cleared release with the `WORK_IN_PROGRESS` state, this state is succesfully set in SW360 and no attachment upload is performed. Afterwards however, if I want to update release information, e.g., download url or clearing state to `OSM_APPROVED`, these changes are not visible in SW360. Do you have an idea what is wrong, or is this somehow an expected SW360 behavior?

**Comment by neubs-bsi** [comment]
@KorSin 
have you set ``sw360updateReleases` to `true` in the updaters properties file? By default it is false.

**Comment by KorSin** [comment]
Yes, it is set to `true`.

File: antenna-documentation/src/site/markdown/sw360-data-model.md.vm [hunk 1]
```diff
@@ -66,8 +66,10 @@ In the following section there is an outline what data of SW360 ${docNameCap} ma
...
-  | SW360 Clearing State | Can not be set with any analyzer, but is a reflection of the clearing state from the SW360 instance data model itself. The SW360 instance clearing state values are `NEW_CLEARING`, `SENT_TO_CLEARING_TOOL`, `REPORT_AVAILABLE`, `APPROVED`. This state is reflected in the SW360 instances UI in the `Clearing State`. |
+  | Clearing State | State reflecting the clearing state within the ${docNameCap} curation processes with the values: `INITIAL`, `EXTERNAL_SOURCE`, `AUTO_EXTRACT`, `PROJECT_APPROVED`, `OSM_APPROVED`, `WORK_IN_PROGRESS`. This is stored in the additional data field with the key `clearingState`. |
+  | SW360 Clearing State | Can not be set with any analyzer, but is a rqeflection of the clearing state from the SW360 instance data model itself. The SW360 instance clearing state values are `NEW_CLEARING`, `SENT_TO_CLEARING_TOOL`, `REPORT_AVAILABLE`, `APPROVED`. This state is reflected in the SW360 instances UI in the `Clearing State`. |
```
Comment (oheger-bosch) [inline antenna-documentation/src/site/markdown/sw360-data-model.md.vm:70]: typo: rqeflection

**Comment by oheger-bosch** [comment]
Do updates of attributes work if you set the clearing state directly to OSM_APPROVED?
And can you update properties multiple times while the state is WORK_IN_PROGRESS?

**Comment by KorSin** [comment]
No, both things do not work. However, I can still add attributes that were not filled out before.

**Comment by oheger-bosch** [comment]
Hm, then this sounds like another problem with updates, but not related to this PR. I would propose adding a new ticket to track this.

**Comment by KorSin** [comment]
Hmm, going through the procedure with a debugger, it seems the problem lies rather on the antenna side:
During an update, the method `getOrCreateRelease` is called. If the release is found in SW360, the data is merged and the update is performed afterwards, see [here](https://github.com/eclipse/antenna/blob/a66aa2e60b17c51d63b97a87cea2e310ace1eb9f/modules/sw360/sw360-workflow/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java#L135). It seems that this merging overwrites the local release with the data from SW360 and then updates it which would make the update redundant. Am I missing something?

**Comment by oheger-bosch** [comment]
Can it be that the merge is just done in the wrong direction? The properties of the local release should override the ones of the release in SW360, and not the way around.

**Comment by KorSin** [comment]
It seems to be the current case. Maybe the intention was to not override the most important fields like the name or version. Changing the priority should not be difficult, but I am unsure whether to do this in this PR.

**Comment by oheger-bosch** [comment]
No, better having a separate ticket and PR. So, I think this one can be merged - maybe except for this typo?

**Comment by neubs-bsi** [comment]
The ticket already exists. https://github.com/eclipse/antenna/issues/588

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**